consensus/istanbul/backend/randao.go

```diff
@@ -118,15 +124,25 @@ func (sb *backend) CalcRandao(number *big.Int, prevMixHash []byte) ([]byte, []by
 	randomReveal := bls.Sign(sb.blsSecretKey, msg[:]).Marshal()
 
 	// calc_mix_hash() = xor(prevMixHash, keccak256(randomReveal))
-	mixHash := calcMixHash(randomReveal, prevMixHash)
+	mixHash := calcMixHash(randomReveal, parentMixHash)
 
 	return randomReveal, mixHash, nil
 }
 
+// VerifyRandao verifies whether header.RandomReveal is the same as expected.
 func (sb *backend) VerifyRandao(chain consensus.ChainReader, header *types.Header, prevMixHash []byte) error {
 	if header.Number.Sign() == 0 {
 		return nil // Do not verify genesis block
 	}
+	if header.RandomReveal == nil || header.MixHash == nil {
+		return errInvalidRandaoFields
+	}
+	// The following if condition is only true when header's block number is the Randao hardfork block number.
+	// Because of the above condition, prevMixHash cannot be nil after Randao hardfork block.
```

Comment:
```suggestion
	// The VerifyRandao is invoked only since Randao hardfork block number.
	// Since Randao hardfork, the header fields are cannot be nil because of the check above (header.RandomReveal == nil || header.MixHash == nil).
	// Therefore it's safe to assume that if prevMixHash == nil, then the given header is exactly Randao hardfork block number.
```